pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -1338,6 +1338,47 @@ func isPhaseStatusComplete(phaseInfo *enterpriseApi.PhaseInfo) bool {
 	}
 }
 
+// validatePhaseInfo validates if phase and status in phaseInfo is valid
+func validatePhaseInfo(ctx context.Context, phaseInfo *enterpriseApi.PhaseInfo) bool {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("validatePhaseInfo").WithValues("phaseInfo", phaseInfo)
+
+	// Check for phase in phaseInfo
+	phases := string(
+		enterpriseApi.PhaseDownload +
+			enterpriseApi.PhasePodCopy +
+			enterpriseApi.PhaseInstall)
+
+	if !strings.Contains(phases, string(phaseInfo.Phase)) {
+		scopedLog.Info("Invalid phase in PhaseInfo")
+		return false
+	}
+
+	// Check for status in phaseInfo
+	statuses := map[enterpriseApi.AppPhaseStatusType]bool{
+		enterpriseApi.AppPkgDownloadPending:     true,
+		enterpriseApi.AppPkgDownloadInProgress:  true,
+		enterpriseApi.AppPkgDownloadComplete:    true,
+		enterpriseApi.AppPkgDownloadError:       true,
+		enterpriseApi.AppPkgPodCopyPending:      true,
+		enterpriseApi.AppPkgPodCopyInProgress:   true,
+		enterpriseApi.AppPkgPodCopyComplete:     true,
+		enterpriseApi.AppPkgMissingFromOperator: true,
+		enterpriseApi.AppPkgPodCopyError:        true,
+		enterpriseApi.AppPkgInstallPending:      true,
+		enterpriseApi.AppPkgInstallInProgress:   true,
+		enterpriseApi.AppPkgInstallComplete:     true,
+		enterpriseApi.AppPkgMissingOnPodError:   true,
+		enterpriseApi.AppPkgInstallError:        true,
+	}
```

Comment:
Instead of filing it on every iteration, probably, we can fill it in a global map, and use that, wherever necessary.